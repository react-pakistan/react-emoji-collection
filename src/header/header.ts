import { IResponsiveProps, responsiveComponent } from '@taimoormk/util-functions';
import styled from 'styled-components';

export const Header = styled.header<IHeaderProps>`

  ${({
    desktop,
    laptop,
    laptopL,
    mobileL,
    mobileM,
    mobileS,
    tablet,
  } : IHeaderProps) : string => responsiveComponent({
    desktop,
    laptop,
    laptopL,
    mobileL,
    mobileM,
    mobileS,
    tablet,
  })};
`;

export interface IHeaderProps extends IResponsiveProps {}
