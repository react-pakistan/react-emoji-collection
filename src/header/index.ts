export {
  Header,
  IHeaderProps,
} from './header';
