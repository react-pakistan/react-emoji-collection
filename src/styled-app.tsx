// tslint:disable typedef

import { LogoMiscColor1 } from '@taimoormk/react-logo-collection/logos/misc-color';
import { theme, ITheme } from '@taimoormk/util-functions';
import * as React from 'react';
import * as styledComponents from 'styled-components';

// https://www.styled-components.com/docs/api#define-a-theme-interface
const {
  default: styled,
  css,
  createGlobalStyle,
  keyframes,
  ThemeProvider,
} = styledComponents as styledComponents.ThemedStyledComponentsModule<ITheme>;

export { styled, css, createGlobalStyle, keyframes, ThemeProvider };

export const StyledApp = styled.div`
  display: inline-block;
  font-family: ${props => props.theme.typography.fontFamily};
  width: 100%;
  *,
  *::before,
  *::after {
    box-sizing: border-box;
  }
`;

export const StyledStory = ({ children } : { children : React.ReactNode }) => (
  <ThemeProvider theme={theme}>
    <StyledApp>{children}</StyledApp>
    <LogoMiscColor1 fontSize="200px" />
  </ThemeProvider>
);

// Due to our `typedef` rule, this type is needed as a helper.
// Actually, by using helpers exported above, styled-components-props
// can be calculated automatically by tsc already.
export type WithTheme<P = {}> = styledComponents.ThemedStyledProps<P, ITheme>;
