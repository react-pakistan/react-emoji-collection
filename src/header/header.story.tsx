import React, { ReactElement } from 'react';
import { StyledStory } from '../styled-app';
import { Header } from '..';

export default {
  title: 'Layout|Header',

  parameters: {
    component: Header,
  },
};

export const HeaderStory = () : ReactElement => (
  <StyledStory>
    <Header>
      Header component
    </Header>
  </StyledStory>
);
