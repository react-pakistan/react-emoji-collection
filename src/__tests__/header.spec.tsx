import React from 'react';
import { mount } from 'enzyme';
import { StyledStory } from '../styled-app';
import { Header } from '../header';

describe('Testing Header component', () => {
  it('matches Header snapshot', () => {
    const wrapper = mount(
      <StyledStory>
        <Header />
      </StyledStory>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
