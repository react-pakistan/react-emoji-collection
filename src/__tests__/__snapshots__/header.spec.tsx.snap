// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header component matches Header snapshot 1`] = `ReactWrapper {}`;
